pifi/videoprocessor.py

```diff
@@ -318,13 +335,22 @@ def __get_process_and_play_vid_cmd(self, ffmpeg_to_python_fifo_name):
             maybe_save_video_tee = shlex.quote(temp_video_save_path) + ' '
             maybe_mv_saved_video_cmd = '&& mv ' + shlex.quote(temp_video_save_path) + ' ' + shlex.quote(video_save_path)
 
+        fps_cmd = f'>( ffprobe -v 0 -of csv=p=0 -select_streams v:0 -show_entries stream=r_frame_rate - >{fps_fifo_name} ) '
         process_and_play_vid_cmd = (
             'set -o pipefail && export SHELLOPTS && ' +
-            vid_data_cmd + "tee " +
+            # Pass --output-error=warn to make sure that early-closed pipes
+            # don't shut down the entire tee. In particular, the fps_cmd will
+            # terminate early, as soon as it discovers the FPS.
+            # Unfortunately, tee is unable to return a zero error code if
+            # any of its output pipes terminate early. So, we're forced to
+            # suppress the non-zero error code with || true. Luckily, the main
+            # process we're interested in the error code of is the yt-dlp one.
+            vid_data_cmd + "( tee --output-error=warn " +
```

Comment:
I believe you can use `--output-error=warn-nopipe` to avoid the exit code jank:
```
% yt-dlp -o - 'https://www.youtube.com/watch?v=YLslsZuEaNE' 2>/dev/null | tee --output-error=warn-nopipe >( ffprobe -v 0 -of csv=p=0 -select_streams v:0 -show_entries stream=r_frame_rate - ) >/dev/null ; echo $?
30/1
0
```

Reply:
Thinking about this more: if we use `--output-error=warn-nopipe` it could obscure an error elsewhere in the pipeline, e.g. in the ffmpeg tee. I think we can work around this via:
```
% yt-dlp -o - 'https://www.youtube.com/watch?v=YLslsZuEaNE' 2>/dev/null | tee >( ffprobe -v 0 -of csv=p=0 -select_streams v:0 -show_entries stream=r_frame_rate - && cat - >/dev/null ) >/dev/null ; echo $?
30/1
0
```